opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/deleteOperation-test.ts

```diff
@@ -0,0 +1,198 @@
+                        id
+                        name
+                    }
+                    main_entity_name
+                    main_entity_type
+                    main_entity_id
+                    deleted_elements {
+                        id
+                        source_index
+                    }
+                }
+            }
+        }
+    }
+`;
+
+const DELETE_CONFIRM_MUTATION = gql`
+    mutation deleteOperationConfirm($id: ID!) {
+        deleteOperationConfirm(id: $id)
+    }
+`;
+
+const DELETE_RESTORE_MUTATION = gql`
+    mutation deleteOperationRestore($id: ID!) {
+        deleteOperationRestore(id: $id)
+    }
+`;
+
+describe('Delete operation resolver testing', () => {
+  let reportInternalId = '';
+  let deleteOperationId = '';
+
+  it('should deleteOperation be created', async () => {
+  // Create and delete the report
+    const REPORT_TO_CREATE = {
+      input: {
+        name: 'Report for deletion',
+        description: 'Report for deletion description',
+        published: '2020-02-26T00:51:35.000Z',
+      },
+    };
+    const report = await queryAsAdmin({ query: CREATE_REPORT_QUERY, variables: REPORT_TO_CREATE });
+    reportInternalId = report.data?.reportAdd.id;
+    await queryAsAdmin({ query: DELETE_REPORT_QUERY, variables: { id: reportInternalId }, });
+
+    // Check that an associated delete operation was created
+    const getAllDeletedOperations = await queryAsAdminWithSuccess({ query: LIST_DELETE_OPERATION_QUERY,
+      variables: {
+        filters: {
+          mode: 'and',
+          filters: [{
+            key: 'main_entity_id',
+            values: reportInternalId,
+            operator: 'eq',
+            mode: 'or',
+          }],
+          filterGroups: [],
+        } } });
+    expect(getAllDeletedOperations.data?.deleteOperations.edges.length).toEqual(1);
+    deleteOperationId = getAllDeletedOperations.data?.deleteOperations.edges[0].node.id;
+
+    const getDeleteOperation = await queryAsAdmin({ query: READ_DELETE_OPERATION_QUERY, variables: { id: deleteOperationId, }, });
+    expect(getDeleteOperation.data?.deleteOperation).toBeDefined();
+    expect(getDeleteOperation.data?.deleteOperation.main_entity_type).toBe(ENTITY_TYPE_CONTAINER_REPORT);
+    expect(getDeleteOperation.data?.deleteOperation.main_entity_id).toBe(reportInternalId);
+    expect(getDeleteOperation.data?.deleteOperation.deleted_elements.length).toBe(1);
+    expect(getDeleteOperation.data?.deleteOperation.deleted_elements[0].id).toBe(reportInternalId);
+  });
+
+  it('should deleteOperation be deleted', async () => {
+    await queryAsAdmin({ query: DELETE_CONFIRM_MUTATION, variables: { id: deleteOperationId }, });
+
+    const queryResult = await queryAsAdminWithSuccess({ query: READ_DELETE_OPERATION_QUERY, variables: { id: deleteOperationId } });
+    expect(queryResult.data?.deleteOperation).toBeNull();
+  });
+
+  it.skip('should deleteOperation be restored', async () => {
```

Comment:
skip ?

Reply:
restore will be implemented in chunk 2 so it needs to be skipped for now, but we already wrote a basic test case for it